app/src/androidTest/java/ch/hikemate/app/ui/map/RunHikeScreenTest.kt

```diff
@@ -166,21 +168,41 @@ class RunHikeScreenTest {
   }
 
   @Test
-  fun runHikeScreen_displaysLoadingAnimationForWaypoints() = runTest {
+  fun runHikeScreen_displaysErrorMessageForWaypoints() = runTest {
     setupCompleteScreenWithSelected(detailedHike, waypointsRetrievalSucceeds = false)
 
+    // So far, the waypoints retrieval should have happened once
+    verify(hikesRepository, times(1)).getRoutesByIds(any(), any(), any())
+
+    // An error message should be displayed to the user, along with a go back action
+    composeTestRule.onNodeWithTag(CenteredErrorAction.TEST_TAG_CENTERED_ERROR_MESSAGE)
     composeTestRule
-        .onNodeWithTag(CenteredLoadingAnimation.TEST_TAG_CENTERED_LOADING_ANIMATION)
+        .onNodeWithTag(CenteredErrorAction.TEST_TAG_CENTERED_ERROR_BUTTON)
         .assertIsDisplayed()
+        .assertHasClickAction()
+        .performClick()
+
+    // Clicking the button should trigger going back with the navigation
+    verify(mockNavigationActions, times(1)).goBack()
   }
 
   @Test
   fun runHikeScreen_displaysLoadingAnimationForElevation() = runTest {
```

Comment:
Nitpick: This test's name is not very descriptive, it think it would be more understandable if it were called `runHikeScreen_displaysLoadingAnimationForElevation_whenElevatrionRetrievalFails`

Reply:
Same as above, settled for a name in the middle: more descriptive and shorter. Let me know what you think!